### PR TITLE
Change OSX to macOS in error message

### PR DESCRIPTION
**Pull request by JamesNK**
**Description**

We mix between the names OSX and macOS in class/variable names in code, but documentation commonly refers to the OS as macOS. Changing this exception message designed for end users to use the common name.

**Customer impact**

Consistent language between documentation and product.

**Regression?**

No

**Risk**

Low

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

**Review by halter73 — APPROVED** [review]

**Review by analogrelay — APPROVED** [review]
This is a certified C O O L change for sure.

**Comment by anurse** [comment]
Did this get ask-mode approval before merging? I probably should have been clearer on the need for an actual sign-off from Tactics.

**Comment by JamesNK** [comment]
Ah, no it didn't. Who should I talk to about that?

**Comment by anurse** [comment]
Since it's been merged, I'll send an email about it now I think it will be OK since it's changing strings. I quickly wrote something up about the process here: https://github.com/aspnet/AspNetCore-Internal/wiki/.NET-Core-3-Ask-Mode

**Pull request merged**
